lib/addoninfo.cpp

```diff
@@ -52,47 +52,70 @@ static std::string parseAddonInfo(AddonInfo& addoninfo, const picojson::value &j
         return "Loading " + fileName + " failed. " + json_error;
     }
     if (!json.is<picojson::object>())
-        return "Loading " + fileName + " failed. Bad json.";
-    picojson::object obj = json.get<picojson::object>();
-    if (obj.count("args")) {
-        if (!obj["args"].is<picojson::array>())
-            return "Loading " + fileName + " failed. args must be array.";
-        for (const picojson::value &v : obj["args"].get<picojson::array>())
-            addoninfo.args += " " + v.get<std::string>();
-    }
+        return "Loading " + fileName + " failed. JSON is not an object.";
 
-    if (obj.count("ctu")) {
-        // ctu is specified in the config file
-        if (!obj["ctu"].is<bool>())
-            return "Loading " + fileName + " failed. ctu must be boolean.";
-        addoninfo.ctu = obj["ctu"].get<bool>();
-    } else {
-        addoninfo.ctu = false;
+    // TODO: remove/complete default value handling for missing fields
+    const picojson::object& obj = json.get<picojson::object>();
+    {
+        const auto it = obj.find("args");
+        if (it != obj.cend()) {
+            const auto& val = it->second;
+            if (!val.is<picojson::array>())
+                return "Loading " + fileName + " failed. 'args' must be an array.";
+            for (const picojson::value &v : val.get<picojson::array>())
+                addoninfo.args += " " + v.get<std::string>();
+        }
     }
 
-    if (obj.count("python")) {
-        // Python was defined in the config file
-        if (obj["python"].is<picojson::array>()) {
-            return "Loading " + fileName +" failed. python must not be an array.";
+    {
```

Comment:
could we implement some convenience functions to read attribute values from a json object.

Reply:
As I mention in the comment - I tried that and t didn't really improve things and will take another stab at it when I adjusted the other code.